flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceTaskTerminationTest.java

```diff
@@ -82,7 +84,7 @@ private void stopWithSavepointStreamTaskTestHelper(final boolean withMaxWatermar
 		final Thread executionThread = srcTaskTestHarness.invoke();
 		final StreamTask<Long, ?> srcTask = srcTaskTestHarness.getTask();
 
-		ready.await();
+		waitForLatchSetByTask(ready, srcTaskTestHarness);
```

Comment:
I did not get the point why we need this change and it seems unrelated to this PR.
If we want to refactor some previous tests, it should be a `hotfix` commit if minor and give some descriptions in commit message for better understanding the motivation.

Reply:
At some point, this test was failing with a timeout 
but It was unclear why do I get this timeout.
I'll extract this to a separate hotfix commit.